File: typescript/src/start-workflows.ts
import { Connection, Client } from '@temporalio/client';
import { greeter } from './workflows';

async function run() {
  const connection = await Connection.connect();

  const client = new Client({
    connection,
  });

  let handle = await client.workflow.start(greeter, {
    taskQueue: 'bugfix-demo',
    args: ["Alice"],
    workflowId: 'bugfix-demo-alice',
  });
  console.log(`Started workflow ${handle.workflowId}.`);

  handle = await client.workflow.start(greeter, {
    taskQueue: 'bugfix-demo',
    args: ["Bob"],
    workflowId: 'bugfix-demo-bob',
  });
  console.log(`Started workflow ${handle.workflowId}.`);
}

run().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: python/start-workflows.py
#!/usr/bin/env python3

import asyncio
import logging
from temporalio.client import Client

from workflow import *

async def main() -> None:
    logging.basicConfig(level=logging.INFO)

    client = await Client.connect("localhost:7233")

    await client.start_workflow(
        Greeter.run,
        "Alice",
        id="bugfix-demo-alice",
        task_queue="bugfix-demo",
    )

    await client.start_workflow(
        Greeter.run,
        "Bob",
        id="bugfix-demo-bob",
        task_queue="bugfix-demo",
    )

if __name__ == "__main__":
    asyncio.run(main())
